experimental/entity/entity.proto

```diff
@@ -0,0 +1,270 @@
+    map<string,string> labels = 3;  
+
+    // NOTE: Although APIs will limit exposing this to people without permissions,
+    // The contents of this map (key+value) may be stored in externally secured object stores (S3, disk, etc)
+    // The values should be lookup keys into a secret service
+    map<string,string> secureKeys = 4;
+
+    // Where did the entity come from
+    Provinance provinance = 5;
+}
+
+// Define how an item got into the system
+message Provinance {
+     // Unix millis when the event happened
+    int64 when = 1;
+
+    // Identifier for the source.  ex: provisioning
+    string source = 2; 
+
+    // optional path to the original source
+    string path = 3; 
+}
+
+// System manages these properties
+message StorageMetadata {
+    // Unix millis for the creation time
+    int64 createdAt = 1;
+
+    // Unix millis for last update time
+    int64 updatedAt = 2;
+
+    // UserID that last updated the entity
+    string updatedBy = 3;
+
+    // Version identifier (might be a hash, does not need to be sortable)
+    string version = 4; 
+
+    // Content checksum
+    string etag = 5; 
+
+    // Size in bytes
+    int64 size = 6;
+
+    // list of GRNs? extracted from the body... based on kind?
+    repeated EntityLocator dependencies = 8;  
+
+    // list of operations the calling user can perform
+    repeated string accessControl = 9; 
+
+    // avaliable PRs for this path
+    repeated EntityPR prs = 10; 
+}
+
+message EntityLocator {
+    string path = 1;
+    string kind = 2;
+}
+
+message EntityPR {
+    // PR Title
+    string name = 2;
+
+    // External URL
+    string url = 1;
+
+    // passed to get and it will load the content
+    string version = 3; 
+}
+  
+message Folder {
+    // uid?  with slashes
+    string path = 1; 
+
+    // v1  (includes container+object)
+    string apiVersion = 3; 
+    
+    // Folder properties
+    EntityProperties props = 4;
+
+    // This refers to the folder propeties, not the contents of the folder
+    StorageMetadata meta = 5; 
+
+    // Child entity information
+    repeated Entity items = 6;
+
+    // More items exist
+    string nextPageToken = 7;
+}
+
+message EntityHistory {
+    // Unix millis when the event happened
+    int64 when = 1; 
+    // Identifier for the source 
+    string who = 2; 
+    // optional path to source location
+    string comment = 3; 
+}
+
+message EntityHistoryResponse {
+    string path = 1; 
+    string kind = 2;
+
+    repeated EntityHistory event = 3;
+
+    // optional path to source location
+    string nextPageToken = 4; 
+}
+
+//---------------------------------------------------------
+// Resource service enables HTTP-style requests over gRPC.
+//---------------------------------------------------------
+
+message AuthInfo {
+    string uid = 1;
+    // teams/groups/role?
+    // ????
+}
+
+message GetEntityRequest {
+    AuthInfo auth = 1;
+
+    string path = 2; 
+    string kind = 3; 
```

Comment:
Do we need the `kind` in the request? I see that `ListFolderRequest` doesn't have it